hack/common.sh

```diff
@@ -132,7 +131,7 @@ get_version() {
   GIT_COMMIT=$(git rev-parse --short HEAD)
   VERSION="${OPERATOR_VERSION}+${GIT_COMMIT}"
 
-  if [ -n "$(git status --porcelain)" ]; then
+  if [ -n "$(git status version tools.go go.mod go.sum vendor Makefile build cmd hack pkg --porcelain)" ]; then
```

Comment:
I dont understand why we are being so specific here, it will make it a pain to update in the future. Can you call out in the code why this needs to be done?

Reply:
Added comment in `dockerfile` and `common.sh`